### PR TITLE
work with newest sqlite jdbc driver and runs faster

### PR DESCRIPTION
**Pull request by tsingakbar**
[https://bitbucket.org/xerial/sqlite-jdbc/downloads](sqlite jdbc 3.15.1)

### CODE REVIEW

File: src/db/SQLiteHelper.java [hunk 1]
```diff
@@ -41,7 +44,7 @@ public SQLiteHelper() {
     public void createDatabase(String databaseName) {
         this.databaseName = databaseName;
         try {
-            connection = DriverManager.getConnection("jdbc:sqlite:");
+            connection = DriverManager.getConnection("jdbc:sqlite:"+databaseName+".db");
```
Comment (tuanna-hsp) [inline src/db/SQLiteHelper.java]: Omit database name when creating connection will make the `SQLite JDBC` to store the new database in RAM, which allows very fast operations (after that we rewrite the content to disk in `writeDatabaseToDisk()`). 
So I dont think adding a database name here is a good choice.

**Review by tuanna-hsp — CHANGES_REQUESTED** [review]

File: src/db/SQLiteHelper.java (hunk 1 above)
Comment (tsingakbar) [reply src/db/SQLiteHelper.java]: oh...I see.
I was converting one dictionary with your program, but the insert operation just endlessly blocked at the 3rd word.
I thought possibly it is your code won't work with my downloaded version of jdbc driver, so I followed jdbc driver's instruction to modify your code, but only to discover that it is your escapeSqlString() doesn't work quite well.
In my opinion, memory table write then flush to disk is a better idea than my batch insert statment version. But other modifications such as preparedstatement should be adopted.
If we can shake on this, I will remove my batch insert statement code when I'm free.

**Comment by tuanna-hsp** [comment]
Ok, that was the only problem. Others were good. Thank for contributing :)

2016/11/28 午後7:45 "Yu Qing" <notifications@github.com>:

**Comment by tsingakbar** [comment]
hey, I revised my code.
Also changed the synonym table codes to use prepared-statement, but I don't have a dictionary with synonym data, so pls help to verify it works for synonym data.

**Comment by tuanna-hsp** [comment]
Thank you for your work, i will make a review later when i have time

2016/11/30 午後9:09 "Yu Qing" <notifications@github.com>:

**Pull request merged**